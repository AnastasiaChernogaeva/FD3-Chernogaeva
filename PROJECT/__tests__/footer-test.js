import React from 'react';
import renderer from 'react-test-renderer';
 
/*import  '../App.js';*/
import Footer from '../components/footer';


test('работа Footer', () => {
 
  // создаём тестовую версию компонента
  const component = renderer.create(
    <Footer />
  );
 
  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
 
  // найдём в вёрстке компонента сами кнопки
 
 
  
  const buttonEleme1 = component.root.find( el =>(el.props.testPr1=="buttOn")); 
  const buttonEleme2 = component.root.find( el =>(el.props.testPr2=="buttOn")); 
  const buttonEleme3 = component.root.find( el =>(el.props.testPr3=="buttOn")); 

  const buttonElems=[buttonEleme1, buttonEleme2, buttonEleme3,];
  // находим все кнопки
 
  buttonElems.forEach(elem=>{elem.props.onClick(); // по очереди нажимаем на кнопки и делаем снимк(так два раза)
  
  // получаем уже изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
  // и "нажмём" на неё
 
  elem.props.onClick();
  
  // получаем уже изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
 
})
})  
 