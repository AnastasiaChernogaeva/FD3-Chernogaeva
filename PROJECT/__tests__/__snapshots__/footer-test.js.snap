// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа Footer 1`] = `
<div
  className="Footer"
>
  <div
    className="F"
  >
    <h2>
      Моя страница
    </h2>
    <p>
      <button
        id="FirstF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           person
        </span>
          Моя страница
      </button>
    </p>
    <p>
      <button
        id="SecF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           shopping_cart
        </span>
          Корзина
      </button>
    </p>
    <p>
      <button
        id="ThF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           favorite
        </span>
          WishList
      </button>
    </p>
  </div>
  <div
    className="S"
  >
    <h2>
      Наши контакты 
    </h2>
    <p>
      <span
        className="material-icons"
      >
        room
      </span>
         г.Черас, проспект Марна,9
    </p>
    <p>
      <span
        className="material-icons"
      >
         call
      </span>
        80336853937
    </p>
    <p>
      <span
        className="material-icons"
      >
         email
      </span>
        sergeychernogaev1979@gmail.com
    </p>
    <p>
      <span
        className="material-icons"
      >
         schedule
      </span>
        7 дней в неделю с 08:00 до 20:30
    </p>
  </div>
</div>
`;

exports[`работа Footer 2`] = `
<div
  className="Footer"
>
  <div
    className="F"
  >
    <h2>
      Моя страница
    </h2>
    <p>
      <button
        id="FirstF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           person
        </span>
          Моя страница
      </button>
    </p>
    <p>
      <button
        id="SecF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           shopping_cart
        </span>
          Корзина
      </button>
    </p>
    <p>
      <button
        id="ThF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           favorite
        </span>
          WishList
      </button>
    </p>
  </div>
  <div
    className="S"
  >
    <h2>
      Наши контакты 
    </h2>
    <p>
      <span
        className="material-icons"
      >
        room
      </span>
         г.Черас, проспект Марна,9
    </p>
    <p>
      <span
        className="material-icons"
      >
         call
      </span>
        80336853937
    </p>
    <p>
      <span
        className="material-icons"
      >
         email
      </span>
        sergeychernogaev1979@gmail.com
    </p>
    <p>
      <span
        className="material-icons"
      >
         schedule
      </span>
        7 дней в неделю с 08:00 до 20:30
    </p>
  </div>
</div>
`;

exports[`работа Footer 3`] = `
<div
  className="Footer"
>
  <div
    className="F"
  >
    <h2>
      Моя страница
    </h2>
    <p>
      <button
        id="FirstF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           person
        </span>
          Моя страница
      </button>
    </p>
    <p>
      <button
        id="SecF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           shopping_cart
        </span>
          Корзина
      </button>
    </p>
    <p>
      <button
        id="ThF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           favorite
        </span>
          WishList
      </button>
    </p>
  </div>
  <div
    className="S"
  >
    <h2>
      Наши контакты 
    </h2>
    <p>
      <span
        className="material-icons"
      >
        room
      </span>
         г.Черас, проспект Марна,9
    </p>
    <p>
      <span
        className="material-icons"
      >
         call
      </span>
        80336853937
    </p>
    <p>
      <span
        className="material-icons"
      >
         email
      </span>
        sergeychernogaev1979@gmail.com
    </p>
    <p>
      <span
        className="material-icons"
      >
         schedule
      </span>
        7 дней в неделю с 08:00 до 20:30
    </p>
  </div>
</div>
`;

exports[`работа Footer 4`] = `
<div
  className="Footer"
>
  <div
    className="F"
  >
    <h2>
      Моя страница
    </h2>
    <p>
      <button
        id="FirstF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           person
        </span>
          Моя страница
      </button>
    </p>
    <p>
      <button
        id="SecF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           shopping_cart
        </span>
          Корзина
      </button>
    </p>
    <p>
      <button
        id="ThF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           favorite
        </span>
          WishList
      </button>
    </p>
  </div>
  <div
    className="S"
  >
    <h2>
      Наши контакты 
    </h2>
    <p>
      <span
        className="material-icons"
      >
        room
      </span>
         г.Черас, проспект Марна,9
    </p>
    <p>
      <span
        className="material-icons"
      >
         call
      </span>
        80336853937
    </p>
    <p>
      <span
        className="material-icons"
      >
         email
      </span>
        sergeychernogaev1979@gmail.com
    </p>
    <p>
      <span
        className="material-icons"
      >
         schedule
      </span>
        7 дней в неделю с 08:00 до 20:30
    </p>
  </div>
</div>
`;

exports[`работа Footer 5`] = `
<div
  className="Footer"
>
  <div
    className="F"
  >
    <h2>
      Моя страница
    </h2>
    <p>
      <button
        id="FirstF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           person
        </span>
          Моя страница
      </button>
    </p>
    <p>
      <button
        id="SecF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           shopping_cart
        </span>
          Корзина
      </button>
    </p>
    <p>
      <button
        id="ThF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           favorite
        </span>
          WishList
      </button>
    </p>
  </div>
  <div
    className="S"
  >
    <h2>
      Наши контакты 
    </h2>
    <p>
      <span
        className="material-icons"
      >
        room
      </span>
         г.Черас, проспект Марна,9
    </p>
    <p>
      <span
        className="material-icons"
      >
         call
      </span>
        80336853937
    </p>
    <p>
      <span
        className="material-icons"
      >
         email
      </span>
        sergeychernogaev1979@gmail.com
    </p>
    <p>
      <span
        className="material-icons"
      >
         schedule
      </span>
        7 дней в неделю с 08:00 до 20:30
    </p>
  </div>
</div>
`;

exports[`работа Footer 6`] = `
<div
  className="Footer"
>
  <div
    className="F"
  >
    <h2>
      Моя страница
    </h2>
    <p>
      <button
        id="FirstF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           person
        </span>
          Моя страница
      </button>
    </p>
    <p>
      <button
        id="SecF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           shopping_cart
        </span>
          Корзина
      </button>
    </p>
    <p>
      <button
        id="ThF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           favorite
        </span>
          WishList
      </button>
    </p>
  </div>
  <div
    className="S"
  >
    <h2>
      Наши контакты 
    </h2>
    <p>
      <span
        className="material-icons"
      >
        room
      </span>
         г.Черас, проспект Марна,9
    </p>
    <p>
      <span
        className="material-icons"
      >
         call
      </span>
        80336853937
    </p>
    <p>
      <span
        className="material-icons"
      >
         email
      </span>
        sergeychernogaev1979@gmail.com
    </p>
    <p>
      <span
        className="material-icons"
      >
         schedule
      </span>
        7 дней в неделю с 08:00 до 20:30
    </p>
  </div>
</div>
`;

exports[`работа Footer 7`] = `
<div
  className="Footer"
>
  <div
    className="F"
  >
    <h2>
      Моя страница
    </h2>
    <p>
      <button
        id="FirstF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           person
        </span>
          Моя страница
      </button>
    </p>
    <p>
      <button
        id="SecF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           shopping_cart
        </span>
          Корзина
      </button>
    </p>
    <p>
      <button
        id="ThF"
        onClick={[Function]}
      >
        <span
          className="material-icons"
        >
           favorite
        </span>
          WishList
      </button>
    </p>
  </div>
  <div
    className="S"
  >
    <h2>
      Наши контакты 
    </h2>
    <p>
      <span
        className="material-icons"
      >
        room
      </span>
         г.Черас, проспект Марна,9
    </p>
    <p>
      <span
        className="material-icons"
      >
         call
      </span>
        80336853937
    </p>
    <p>
      <span
        className="material-icons"
      >
         email
      </span>
        sergeychernogaev1979@gmail.com
    </p>
    <p>
      <span
        className="material-icons"
      >
         schedule
      </span>
        7 дней в неделю с 08:00 до 20:30
    </p>
  </div>
</div>
`;
